book/src/application.md

```diff
@@ -0,0 +1,75 @@
+# Building an Application
+
```

Comment:
I think I'm missing a rough overview what this chapter contains. A short intro of what the goal of the chapter is and what the reader can expect.

Based on the title I was expecting a step by step guide to writing new applications, but after reading it seems like a collection of RIOT-rs specifics to writing applications. Setting some expectations in a intro paragraph could help here